martian.go

```diff
@@ -68,18 +62,41 @@ func HTTPRequestExecutor(result *parse.Result, re proxy.HTTPRequestExecutor) pro
 			err = ErrEmptyResponse
 			return
 		}
-		if resp.Body == nil {
-			resp.Body = ioutil.NopCloser(bytes.NewBufferString(""))
-		}
-		if resp.Header == nil {
-			resp.Header = http.Header{}
+		if err = modifyResponse(result.ResponseModifier(), resp); err != nil {
```

Comment:
the error evaluation is not required here since we are going to return the error anyway...